source/administration/data-retention.rst

```diff
@@ -0,0 +1,73 @@
+Data Retention Policy Beta (E20)
+=================================
+
+Available in `Enterprise Edition E20 <https://about.mattermost.com/pricing/>`_.
+
+By default, Mattermost stores all message history providing an unlimited search history to admins and end users.
+
+In Enterprise Edition E20, you may set a custom policy for how long messages and file uploads are kept in Mattermost channels and direct messages.
+
+.. warning:: Once a message or a file is deleted, the action is irreversible. Please be careful when setting up a custom data retention policy.
+
+.. toctree::
+    :maxdepth: 2
+
+Set Up Guide
+--------------
+
+To set a custom data retention policy:
+
+1. Go to **System Console > Advanced > Data Retention Policy (Beta)**
+2. Set your desired retention policy for messages. Messages, including file attachments, older than the duration you set will be deleted nightly. The minimum message retention time is one day.
+3. Similarly, set your desired retention policy for file uploads. File uploads older than the duration you set will be deleted nightly from your file storage system, either from your local disk or your Amazon S3 service as specified in **System Console > Files > Storage**. The minimum file retention time is one day.
```

Comment:
Does the file retention number have to be < the message retention number? Since messages include file attachments, it would be meaningless to set file retention > message retention?

Reply:
It can be longer -- If you have message retention = 7 days, and file retention = 14 days, then
 - messages are deleted from the DB after 7 days
 - messages (including files) are deleted from the UI after 7 days
 - files are deleted from file storage after 14 days.

^Let me know if this should be made more clear though